src/circuit/index.ts

```diff
@@ -1,135 +1,51 @@
-import { logger } from '@libp2p/logger'
-import { codes } from '../errors.js'
-import {
-  setDelayedInterval,
-  clearDelayedInterval
-// @ts-expect-error set-delayed-interval does not export types
-} from 'set-delayed-interval'
-import { AutoRelay } from './auto-relay.js'
-import { namespaceToCid } from './utils.js'
-import {
-  RELAY_RENDEZVOUS_NS
-} from './constants.js'
-import type { AddressSorter, PeerStore } from '@libp2p/interface-peer-store'
-import type { Startable } from '@libp2p/interfaces/startable'
-import type { ContentRouting } from '@libp2p/interface-content-routing'
-import type { ConnectionManager } from '@libp2p/interface-connection-manager'
-import type { TransportManager } from '@libp2p/interface-transport'
-import type { PeerId } from '@libp2p/interface-peer-id'
-import type { StreamHandlerOptions } from '@libp2p/interface-registrar'
-
-export interface RelayConfig extends StreamHandlerOptions {
+/**
+ * RelayConfig configures the circuit v2 relay transport.
+ */
+export interface RelayConfig {
+  /**
+   * Enable dialing a client over a relay and receiving relayed connections.
+   * This in itself does not enable the node to act as a relay.
+   */
   enabled: boolean
   advertise: RelayAdvertiseConfig
   hop: HopConfig
-  autoRelay: AutoRelayConfig
+  reservationManager: RelayReservationManagerConfig
 }
 
-export interface HopConfig {
-  enabled?: boolean
-  active?: boolean
-  timeout: number
-}
-
-const log = logger('libp2p:relay')
-
-export interface RelayAdvertiseConfig {
-  bootDelay?: number
-  enabled?: boolean
-  ttl?: number
-}
-
-export interface AutoRelayConfig {
-  enabled?: boolean
-
+/**
+ * RelayReservationManagerConfig allows the node to automatically listen
+ * on any discovered relays upto a specified maximum.
+ */
+export interface RelayReservationManagerConfig {
   /**
-   * maximum number of relays to listen
+   * enable or disable autorelay (default: false)
    */
-  maxListeners: number
-}
-
-export interface RelayInit extends RelayConfig {
-  addressSorter?: AddressSorter
-}
-
-export interface RelayComponents {
-  peerId: PeerId
-  contentRouting: ContentRouting
-  peerStore: PeerStore
-  connectionManager: ConnectionManager
-  transportManager: TransportManager
-}
-
-export class Relay implements Startable {
-  private readonly components: RelayComponents
-  private readonly init: RelayInit
-  // @ts-expect-error this field isn't used anywhere?
-  private readonly autoRelay?: AutoRelay
-  private timeout?: any
-  private started: boolean
+  enabled?: boolean
```

Comment:
Under what circumstances would a user want to disable the reservation manager?

Are there tests for when it's disabled to ensure libp2p can run without it?

Reply:
We would want to disable the reservation manager when we do not want to automatically listen on relays. It replaces the old name `AutoRelay`. https://github.com/libp2p/js-libp2p/pull/1533#discussion_r1086978369.

The relay tests disable this by default https://github.com/libp2p/js-libp2p/pull/1533/files#diff-81971332c1841fd8324647bd37da55f446e332af9e07d4f28632d5768302dac9R27